dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/engine/command/impl/CreatePrivateUrlCommandTest.java

```diff
@@ -123,28 +123,30 @@ public void testAlreadyExists() {
     }
 
     @Test
-    public void testAttemptCreatePrivateUrlOnNonDraft() {
+    public void testCreatePrivateUrlSuccessfully_onReleased() {
         dataset = new Dataset();
         List<DatasetVersion> versions = new ArrayList<>();
         DatasetVersion datasetVersion = new DatasetVersion();
         datasetVersion.setVersionState(DatasetVersion.VersionState.RELEASED);
         versions.add(datasetVersion);
         dataset.setVersions(versions);
         dataset.setId(latestVersionIsNotDraft);
-        String expected = "Can't create Private URL because the latest version of dataset id " + latestVersionIsNotDraft + " is not a draft.";
-        String actual = null;
-        PrivateUrl privateUrl = null;
-        try {
-            privateUrl = testEngine.submit(new CreatePrivateUrlCommand(null, dataset));
-        } catch (CommandException ex) {
-            actual = ex.getMessage();
-        }
-        assertEquals(expected, actual);
-        assertNull(privateUrl);
+
+        PrivateUrl privateUrl = testEngine.submit(new CreatePrivateUrlCommand(null, dataset));
+        assertNotNull(privateUrl);
+        assertNotNull(privateUrl.getDataset());
+        assertNotNull(privateUrl.getRoleAssignment());
+        PrivateUrlUser expectedUser = new PrivateUrlUser(dataset.getId());
+        assertEquals(expectedUser.getIdentifier(), privateUrl.getRoleAssignment().getAssigneeIdentifier());
+        assertEquals(expectedUser.isSuperuser(), false);
+        assertEquals(expectedUser.isAuthenticated(), false);
+        assertEquals(expectedUser.getDisplayInfo().getTitle(), "Private URL Enabled");
+        assertNotNull(privateUrl.getToken());
+        assertEquals("https://dataverse.example.edu/privateurl.xhtml?token=" + privateUrl.getToken(), privateUrl.getLink());
```

Comment:
Suggestion: in case of so many assertions it would be best to organize them somehow, for example by putting them in separate methods. Alternatively assertion messages, with some explanations, could be added.